### PR TITLE
Add cache option

### PR DESCRIPTION
**Pull request by kzima**
Allow signal events to update config on the fly.

### CODE REVIEW

**Comment by maxbeatty** [comment]
Can you expand the description? I'm not clear on what the use case is

**Comment by kzima** [comment]
Imagine a situation where API key in .env file needs to be changed, but we don't want to restart the node server (this can be achieved with signals). Line 33 in lib/main.js prevents from doing this. So instead of completely removing first expression from `process.env[key] || parsedObj[key]` I just  added an extra cache option.

**Comment by maxbeatty** [comment]
Sorry, I still don't understand how this works. How are you re-running `config` with a signal? Can you provide some example code here or in a gist?

**Comment by kzima** [comment]
// listen to the process signal and restart the environment
process.on('SIGINT', () => {
  require('dotenv').config({ cache: false });
});

**Comment by maxbeatty** [comment]
Can you add a test? This looks to function effectively the same as #130 😬 (`cache` set to `false` would allow overriding the current `process.env`)

**Comment by kzima** [comment]
hey @maxbeatty I have added a test.

File: test/main.js [hunk 1]
```diff
@@ -166,5 +166,15 @@ describe('dotenv', function () {
       parsed.INCLUDE_SPACE.should.eql('some spaced out string')
       done()
     })
+
+    it('should override basic environment if cache is set to false', function (done) {
+      var testPath = 'test/.env'
+      dotenv.config({path: testPath})
+      process.env.BASIC.should.eql('basic')
+      process.env.BASIC = 'test'
+      dotenv.config({path: testPath, chache: false})
```
Comment (jcblw) [inline test/main.js]: hmm is this test passing? If so it might be evergreen cause cache is spelled wrong.

Comment (maxbeatty) [inline test/main.js]: for whatever reason, travis did not build this commit

**Comment by maxbeatty** [comment]
Thanks for the idea @kzima. I'm going to close this in favor of #130 since that's where more discussion about overriding is happening.

**Pull request closed without merging**